include/replayio-macros.h

```diff
@@ -16,6 +16,12 @@
     recordreplay::Assert(format, ##__VA_ARGS__); \
   static_assert(true, "require semicolon")
 
+// Same as |REPLAY_ASSERT| but won't Assert if the given condition is not true.
+#define REPLAY_ASSERT_IF(cond, format, ...) \
```

Comment:
Can these be defined in terms of one another?

e.g. `REPLAY_ASSERT_IF` could be:
```
#define REPLAY_ASSERT_IF(cond, format, ...) \
  if (cond) REPLAY_ASSERT(format, ##__VA_ARGS__)
```

and `REPLAY_ASSERT_MAYBE_EVENTS_DISALLOWED`  could be:

```
#define REPLAY_ASSERT_MAYBE_EVENTS_DISALLOWED(format, ...) \
  REPLAY_ASSERT_IF(!recordreplay::AreEventsDisallowed(), ##__VA_ARGS__);
```

or `REPLAY_ASSERT_IF` could be the one that's ultimately evaluated with `REPLAY_ASSERT` being defined as:

```
#define REPLAY_ASSERT(format, ...) \
  REPLAY_ASSERT_IF(true, format, ## __VA_ARGS__)
```

otherwise I'd be concerned about things diverging between the macros.

Reply:
1. ❌Your first suggestion is a no-go, because suddenly you are evaluating `cond` even when things are enabled.
2. 🤷‍♀️ I don't like the unnecessary `if (true)` in your third suggestion.
3. ✅ I'm not concerned at all, since this is is going to be a very small set of often-used macros.